tests/test_ui_components.py

```
@@ -0,0 +1,336 @@
+from src.ui.components import setup_avatar
+
+
+class TestSetupAvatar:
+    """Test cases for setup_avatar function."""
+
+    @patch('src.ui.components.requests.get')
+    @patch('src.ui.components.Image.open')
+    @patch('src.ui.components.Image.new')
+    @patch('builtins.open', new_callable=mock_open)
+    def test_setup_avatar_successful_download(
+        self, mock_file, mock_image_new, mock_image_open, mock_requests_get
```

Ruff (0.14.8)

Unused method argument: `mock_image_new`

(ARG002)